### PR TITLE
Move TensorBoard and TensorFlow Events UI rendering to Python function to deprecate magic.

### PR DESCRIPTION
**Pull request by qimingj**
- Move TensorBoard and TensorFlow Events UI rendering to Python function to deprecate magic.

- Use matplotlib for tf events plotting so it can display well in static HTML pages (such as github).
    
- Improve TensorFlow Events list/get APIs.

### CODE REVIEW

File: datalab/mlalpha/_summary.py [hunk 1]
```diff
@@ -1,107 +1,127 @@
...
+    return ret_events
+
+  def plot(self, event_names, x='step'):
+    """Plots a list of events. Each event (a dir+event_name) is represetented as a line
+       in the graph.
+
+    Args:
+      event_names: A list of events to plot. Each event_name may correspond to multiple events,
+          each in a different directory.
+      x: whether to use step or time as x.
```
Comment (brandondutra) [inline datalab/mlalpha/_summary.py]: Maybe rename to "x_axis_label" or "x_axis"?

File: datalab/mlalpha/_summary.py [hunk 2]
```diff
@@ -1,107 +1,127 @@
...
+    """
+    event_names = [event_names] if isinstance(event_names, basestring) else event_names
+    events_list = self.get_events(event_names)
+    for event_name, dir_event_dict in zip(event_names, events_list):
+      for dir, df in dir_event_dict.iteritems():
+        label = event_name + ':' + dir
+        x_column = df['step'] if x == 'step' else df['time']
+        plt.plot(x_column, df['value'], label=label)
+        plt.legend(loc='best')
```
Comment (brandondutra) [inline datalab/mlalpha/_summary.py]: does plt.legend() needs to be called in every loop?

File: datalab/mlalpha/_summary.py [hunk 3]
```diff
@@ -1,107 +1,127 @@
...
-    else:
-      path_pattern = os.path.join(self._path, '*tfevents*')
-      return glob.glob(path_pattern)
+    self._paths = [paths] if isinstance(paths, basestring) else paths
+
+  def _glob_events_files(self, paths):
+    event_files = []
+    for path in paths:
+      if path.startswith('gs://'):
+        event_files += ml.util._file.glob_files(os.path.join(path, '*.tfevents.*'))
```
Comment (qimingj) [inline datalab/mlalpha/_summary.py]: Thanks for the pointer. Indeed.
We have many references on google.cloud.ml.util._file.*, which need to be replaced all. I will tackle all in once.

File: datalab/mlalpha/_summary.py (hunk 1 above)
Comment (qimingj) [inline datalab/mlalpha/_summary.py]: Done.

File: datalab/mlalpha/_summary.py (hunk 2 above)
Comment (qimingj) [inline datalab/mlalpha/_summary.py]: No. Done!

**Pull request merged**